src/ftpclass.cc

```diff
@@ -829,6 +827,45 @@ Ftp::pasv_state_t Ftp::Handle_EPSV()
    return PASV_HAVE_ADDRESS;
 }
 
+Ftp::pasv_state_t Ftp::Handle_EPSV_CEPR()
+{
+   char *format=alloca_strdup("(|%u|%[^'|']|%u|)");
+   unsigned port, proto;
+   char pasv_addr[40];
+
+   const char *c_start=strchr(line,'(');
+   if(sscanf(c_start,format,&proto, &pasv_addr, &port)!=3)
```

Comment:
stack overrun possible here, tune the sscanf format

Reply:
Ok. My format now limits the pasv address to be at most 39 chars.